unit/java_bytecode/java_bytecode_parse_generics/parse_generic_functions.cpp

```diff
@@ -17,59 +17,1171 @@
+          REQUIRE(generic_param_t.type_variable().get_identifier() ==
+            class_prefix + "::T");
+          THEN("The bounds are set correctly")
+          {
+            // TODO: the bounds are not parsed yet; extend the tests when fixed
+          }
+        }
+      }
+      THEN("It contains parameter u pointing to Generic")
+      {
+        code_typet::parametert param_u =
+          require_type::require_parameter(func_code, "u");
+        require_type::require_pointer(
+          param_u.type(), symbol_typet("java::Generic"));
+
+        THEN("u is generic with type variable U")
+        {
+          REQUIRE(is_java_generic_type(param_u.type()));
+
+          const java_generic_typet generic_u =
+            to_java_generic_type(param_u.type());
+          const java_generic_parametert generic_param_u =
+            generic_u.generic_type_variables().front();
+          REQUIRE(generic_param_u.type_variable().get_identifier() ==
+            class_prefix + "::U");
+          THEN("The bounds are set correctly")
+          {
+            // TODO: the bounds are not parsed yet; extend the tests when fixed
+          }
+        }
+      }
+    }
+  }
+
+  WHEN("Parsing returnSimple")
+  {
+    THEN("There should be a symbol for the function")
+    {
+      const std::string func_name = ".returnSimple";
+      const std::string func_descriptor = ":()LGeneric;";
+      const std::string process_func_name =
+        class_prefix + func_name + func_descriptor;
+
+      REQUIRE(new_symbol_table.has_symbol(process_func_name));
+
+      const symbolt func_symbol =
+        new_symbol_table.lookup_ref(process_func_name);
+      THEN("It has return type pointing to Generic")
+      {
+        const code_typet func_code = to_code_type(func_symbol.type);
+        const typet return_type = func_code.return_type();
+        require_type::require_pointer(
+          return_type, symbol_typet("java::Generic"));
+
+        THEN("It is generic with type variable T")
+        {
+          REQUIRE(is_java_generic_type(return_type));
+
+          const java_generic_typet generic_return_type =
+            to_java_generic_type(return_type);
+          const java_generic_parametert generic_return_param =
+            generic_return_type.generic_type_variables().front();
+          REQUIRE(generic_return_param.type_variable().get_identifier() ==
+            class_prefix + "::T");
+        }
+      }
+    }
+  }
+
+  WHEN("Parsing returnSimpleField")
+  {
+    THEN("There should be a symbol for the function")
+    {
+      const std::string func_name = ".returnSimpleField";
+      const std::string func_descriptor = ":()Ljava/lang/Object;";
+      const std::string process_func_name =
+        class_prefix + func_name + func_descriptor;
+
+      REQUIRE(new_symbol_table.has_symbol(process_func_name));
+
+      const symbolt func_symbol =
+        new_symbol_table.lookup_ref(process_func_name);
+      // TODO: should point to something else than Object??
```

Comment:
Is this TODO something that should be extended now, or something that we are waiting to be fixed elsewhere before extending? Might be nice to clarify this comment, and add refs to Jira if appropriate.

Reply:
I think it relates to bounds as well, I added a reference to the JIRA task I created for bounds.